locust/test/test_main.py

```diff
@@ -608,24 +842,30 @@ def test_autostart_wo_run_time(self):
                 stderr=PIPE,
                 text=True,
             )
-            gevent.sleep(1.9)
-            try:
-                response = requests.get(f"http://localhost:{port}/")
-            except Exception:
-                pass
+
+            poll_until(lambda: web_interface_ready("localhost", port))
+
+            response = requests.get(f"http://localhost:{port}/")
             self.assertEqual(200, response.status_code)
+
             proc.send_signal(signal.SIGTERM)
-            stdout, stderr = proc.communicate()
+            stdout, stderr = proc.communicate(timeout=5)
+
             self.assertIn("Starting Locust", stderr)
             self.assertIn("No run time limit set, use CTRL+C to interrupt", stderr)
             self.assertIn("Shutting down ", stderr)
             self.assertNotIn("Traceback", stderr)
+
             # check response afterwards, because it really isn't as informative as stderr
             d = pq(response.content.decode("utf-8"))
 
             self.assertEqual(200, response.status_code)
             self.assertIn('"state": "running"', str(d))
 
+            if proc.poll() is None:
```

Comment:
You've already done proc.communicate, so process is guaranteed to be dead already.